tests/unit/script/htmlimageelement.rs

```diff
@@ -0,0 +1,81 @@
+use script::test::String;
+use script::test::srcset::{Descriptor, ImageSource, parse_a_srcset_attribute};
+
+#[test]
+fn no_value() {
+    let new_vec = Vec::new();
+    assert_eq!(parse_a_srcset_attribute(String::new()), new_vec);
+}
+
+#[test]
+fn width_one_value() {
+    let first_descriptor = Descriptor { wid: Some(320), den: None };
+    let first_imagesource = ImageSource { url: "small-image.jpg".to_string(), descriptor: first_descriptor };
+    let mut sources = Vec::new();
+    sources.push(first_imagesource);
```

Comment:
All of these tests can use `let sources = &[first_imagesource];` instead of pushing entries individually.